lib/core/macros.nim

```diff
@@ -1613,6 +1613,15 @@ macro getCustomPragmaVal*(n: typed, cp: typed{nkSym}): untyped =
   if result.kind == nnkEmpty:
     error(n.repr & " doesn't have a pragma named " & cp.repr()) # returning an empty node results in most cases in a cryptic error,
 
+when defined(nimMacrosIsUnion):
+  proc isUnion*(arg: NimNode): bool {.magic: "NIsUnion", noSideEffect, since: (1,1).}
```

Comment:
* seems a better fit in typetraits.nim
* and instead the API should be:
```nim
proc isUnion*(t: typedesc): bool {.magic: "TypeTrait".}
```
(no need for `macro isUnion*(arg: typed)` plus `proc isUnion*(arg: NimNode): bool`)